src/AppInstallerCLICore/ContextOrchestrator.cpp

```diff
@@ -59,6 +59,13 @@ namespace AppInstaller::CLI::Execution
         }
     }
 
+    void ContextOrchestrator::RequeueItem(OrchestratorQueueItem& item)
+    {
+        std::lock_guard<std::mutex> lockQueue{ m_queueLock };
+
+        item.SetState(OrchestratorQueueItemState::Queued);

```

Comment:
>item.SetState(OrchestratorQueueItemState::Queued); [](http://example.com/codeflow?start=8&length=50)

Just a note here for when you do try to enable multiple downloads - one case my code didn't attempt to handle is that it doesn't create a new thread when doing a requeue when it maybe should.
Example scenario:
Simultaneous download limit is 4 and 4 apps are downloading while 2 other apps are waiting.
One of the apps finishes downloading and is requeued. Now that one of the apps is done downloading it probably should create a new thread to do another download to keep it at the 4 download limit while the current thread is installing. If the code doesn't create a new thread here then once the simultaneous download stuff comes in you can end up in a case where the queue looks like:
1. Installing, 2. Downloaded but waiting for Install spot, 3. Downloaded but waiting for Install spot, 4. Downloaded but waiting for install spot, 5. Not started, 6. Not Started.
which seems wrong.

Reply:
Thanks! I hadn't thought of that case. I'll keep it in mind.

Reply:
Ah actually I got it wrong. You can't end up with that case above. The worst you can get is three apps downloading while one installs instead of four downloading while one installs, since the threads that see that the install is blocked don't just sit there waiting they'd move on to trying to do a download for the other item. Depending on the download limit that may be ok to just sometimes be doing one less than the max number of downloads since the system will still be busy doing an install (which i think for some installers also involves some downloading). But yeah, something to add a comment about probably on which behavior gets chosen.